src/dmd/dmodule.d

```diff
@@ -535,6 +501,77 @@ extern (C++) final class Module : Package
         return m;
     }
 
+    /**
+     * Converts a chain of identifiers to the filename of the module
+     *
+     * Params:
+     *  packages = the names of the "parent" packages
+     *  ident = the name of the child package or module
+     *
+     * Returns:
+     *  the filename of the child package or module
+     */
+    static const(char)* getFilename(Identifiers* packages, Identifier ident)
+    {
+        const(char)[] filename = ident.toString();
+
+        if (packages == null || packages.dim == 0)
+            return filename.ptr;
+
+        OutBuffer buf;
+        OutBuffer dotmods;
+        auto ms = global.params.modFileAliasStrings;
+        const msdim = ms ? ms.dim : 0;
```

Comment:
Fix `ms` so it is a `const(char)*[]` and then the awkward `msdim` won't be needed.